### DISCUSSION TITLE
Come gestire le fatture elettroniche rifiutate dallo sdi

### DISCUSSION THREAD

**Discussion by vitoo22** [fattureincloud/api#125] [Q&A] [answered]
Buongoirno.

Sto cercando di capire come vanno gestite le fatture elettroniche che vengono rifiutate dallo sdi una volta inviate, e ho le seguenti domande:

1. Al momento quando invio una fattura elettronica, mi arriva una notifica frontend sia sul sito che via app. E' possibile avere quella notifica via API?
2. Io ho il seguente schema logico nella mia app, che è sostanzialmente una piattaforma ordini per gli agenti che poi usa l'API di fattureincloud v2: inserimento dati attività > inserimento ordine > pagamento > creazione fattura (con dati attività inseriti) > invio fattura elettronica. Se la fattura elettronica viene rifiutata, avvisiamo l'agente che modifica i dati sbagliati dell'attività (di solito il problema è quello). A questo punto, quando l'agente modifica i dati dell'attività, ho bisogno di andare a filtrare le fatture (con relativa fattura elettronica) che sono state rifiutate. C'è un modo suggerito per farlo agevolmente?

Sostanzialmente, la mia necessità è quella di intervenire modificando fatture già create, ma non accettate dallo SDI, in che si riduce ad aver bisogno di filtrare gli "IssuedDocuments" ed avere solo quelli con fattura elettronica rifiutata.

Grazie,
Vittorio

**Comment by emajo** [comment, marked as answer]
Ciao @vitoo22, per quanto riguarda il primo punto stiamo sviluppando dei webhooks che serviranno proprio per questo scopo, per il secondo punto putroppo attualmente non è possibile filtrare sul campo `ei_status`, sarebbe sicuramente utile farlo e cercheremo di introdurre questa funzionalità in futuro, per ora è necessario che vengano filtrati lato vostro.

Grazie,
Emanuele

**Reply by vitoo22** [reply]
Ok Emanuele, grazie per la risposta.
Sicuramente essere in grado di intercettare la risposta/notifica del fattura elettronica diventerebbe molto utile, quindi tengo d'occhio.

Mi sono appena accorto, grazie al tuo messaggio, che in effetti c'e' ei_status e posso comunque controllarle io dal mio lato. Però non ho trovato nella documentazione l'elenco degli ei_status possibili. Sono scarso io a cercare o non è presente questa spiegazione?
Grazie!

**Reply by emajo** [reply]
Ciao @vitoo22 per vedere i possibili valori dei campi enum ti basterà usare il tasto `multiline description` che trovi nella nostra API Reference all'inizio di ogni richiesta/risposta.

Fammi sapere se hai risolto,
grazie,
Emanuele
